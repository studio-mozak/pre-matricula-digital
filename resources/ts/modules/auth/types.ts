export interface Auth {
  level: number;
  name: string;
  schools: string[];
}
