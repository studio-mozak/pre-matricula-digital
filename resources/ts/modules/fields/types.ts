import { Nullable } from '@/types';

export type SetValues = (fields: Record<string, unknown>) => void;

export interface FieldPageOption {
  id?: string;
  name: Nullable<string>;
  weight: number;
}

export interface FieldPage {
  id: Nullable<string>;
  name: Nullable<string>;
  description: Nullable<string>;
  group: Nullable<'RESPONSIBLE' | 'STUDENT'>;
  type: Nullable<string>;
  internal: boolean;
  options: FieldPageOption[];
  deleteOptions?: string[];
  required?: boolean;
}

export interface ErrorResponse {
  message: string;
}
