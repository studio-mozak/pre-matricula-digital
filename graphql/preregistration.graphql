extend type Query @guard(with: ["prematricula"]) {

  """
  Lista as pré-matrículas.
  """
  preregistrations(
    """
    Retorna apenas as pré-matrículas onde o **protocolo** ou o **nome do(a) aluno(a)** contenha `search`.
    """
    search: String @slug @search(columns: ["protocol", "student.slug"])
    process: ID @eq(key: "process_id")
    processes: [ID!] @in(key: "process_id")
    school: ID @eq(key: "school_id")
    schools: [ID!] @in(key: "school_id")
    grade: ID @eq(key: "grade_id")
    grades: [ID] @in(key: "grade_id")
    period: ID @eq(key: "period_id")
    type: PreRegistrationType @eq(key: "preregistration_type_id")
    status: PreRegistrationStatus @eq(key: "status")
    sort: PreRegistrationSort @scope(name: "sort")
    year: Int @search(columns: ["process.school_year_id"])
  ): [PreRegistration!]! @paginate

}

extend type Query @guard(with: ["web", "sanctum", "prematricula"]) {

  """
  Retorna a pré-matrícula do protocolo.
  """
  preregistrationByProtocol(
    protocol: String @eq
    withVacancy: Boolean
  ): PreRegistration @first

  """
  Retorna a inscrição usando o código de verificação.
  """
  preregistrationByCode(
    code: String @eq
  ): PreRegistration @first

  getStudentMatches(
    stage: ID!
    name: String @slug
    dateOfBirth: Date @rename(attribute: "date_of_birth")
    cpf: String
    rg: String
    birthCertificate: String @rename(attribute: "birth_certificate")
  ): [StudentMatch!]!

  summonNextInLine(
    process: ID! @eq(key: "process_id")
    school: ID! @eq(key: "school_id")
    grade: ID! @eq(key: "grade_id")
    period: ID! @eq(key: "period_id")
  ): PreRegistration @first(scopes: ["nextInLine", "waiting"])

  matchByStudentCpf(
    cpf: String! @eq
    year: Int! @eq
  ): RegistrationMatch @first

  matchByStudentRg(
    rg: String! @eq
    year: Int! @eq
  ): RegistrationMatch @first

  matchByStudentBirthCertificate(
    birthCertificate: String! @eq(key: "birth_certificate")
    year: Int! @eq
  ): RegistrationMatch @first

  matchByStudentNameAndDateOfBirth(
    name: String! @slug @eq(key: "slug")
    dateOfBirth: Date! @eq(key: "date_of_birth")
    year: Int! @eq
  ): RegistrationMatch @first

  findPreRegistrationsByStudentCpf(
    cpf: String!
  ): [PreRegistration!]!

  findPreRegistrationsByStudentRg(
    rg: String!
  ): [PreRegistration!]!

  findPreRegistrationsByStudentBirthCertificate(
    birthCertificate: String! @rename(attribute: "birth_certificate")
  ): [PreRegistration!]!

  findPreRegistrationsByStudentNameAndDateOfBirth(
    name: String! @slug
    dateOfBirth: Date! @rename(attribute: "date_of_birth")
  ): [PreRegistration!]!

  findPreRegistrationsByInfo(
    cpf: String @rename(attribute: "responsible_cpf")
    email: String @rename(attribute: "responsible_email")
    protocol: String
  ): [PreRegistration!]!

  getPreRegistrationLinkedByEmail(
    email: String! @eq
  ): [PreRegistrationLinkedByEmail!]! @all

  searchLegacyStudentsByName(
    name: String!
    first: Int
  ): [LegacyStudent!]!
}

enum PreRegistrationType {
  """
  Rematrícula
  """
  REGISTRATION_RENEWAL @enum(value: 1)
  """
  Matrícula
  """
  REGISTRATION @enum(value: 2)
  """
  Lista de espera
  """
  WAITING_LIST @enum(value: 3)
}

enum PreRegistrationStatus {
  """
  Pré-matrícula aguardando (in)deferimento.
  """
  WAITING @enum(value: 1)

  """
  Pré-matrícula deferida.
  """
  ACCEPTED @enum(value: 2)

  """
  Pré-matrícula indeferida.
  """
  REJECTED @enum(value: 3)

  """
  Pré-matrícula convocada.
  """
  SUMMONED @enum(value: 4)

  """
  Pré-matrícula em confirmação.
  """
  IN_CONFIRMATION @enum(value: 5)
}

enum RelationType {
  """
  Mãe
  """
  MOTHER @enum(value: 1)

  """
  Pai
  """
  FATHER @enum(value: 2)

  """
  Responsável legal
  """
  GUARDIAN @enum(value: 3)

  """
  A própria pessoa
  """
  SELF @enum(value: 4)
}

enum PreRegistrationSort {
  DATE @enum(value: 1)
  POSITION @enum(value: 2)
  SCHOOL @enum(value: 3)
  NAME @enum(value: 4)
  DATE_OF_BIRTH @enum(value: 5)
}

type RegistrationMatch {
  id: ID!
  year: Int!
  school: ID!
  grade: ID!
  period: ID!
  initials: String!
  dateOfBirth: Date! @rename(attribute: "date_of_birth")
}

enum StudentMatchType {
  PRE_REGISTRATION_ALREADY_EXISTS @enum(value: 1)
  REGISTRATION_CURRENT_YEAR @enum(value: 2)
  REGISTRATION_LAST_YEAR @enum(value: 3)
  STUDENT_ALREADY_EXISTS @enum(value: 4)
}

type StudentMatch {
  id: ID!
  initials: String!
  dateOfBirth: Date! @rename(attribute: "date_of_birth")
  type: ProcessStageRestriction!
  process: Process
  registration: StudentMatchRegistration
}

type StudentMatchRegistration {
  year: Int!
  school: School!
  grade: Grade!
  period: Period
}

type PreRegistration {
  id: ID!
  type: PreRegistrationType! @rename(attribute: "preregistration_type_id")
  position: Int!
  parent: PreRegistration @belongsTo
  waiting: PreRegistration @belongsTo
  others: [PreRegistration]
  protocol: String!
  code: String!
  date: DateTime! @rename(attribute: "created_at")
  process: Process @belongsTo
  period: Period @belongsTo
  school: School @belongsTo
  grade: Grade @belongsTo
  schoolYear: SchoolYear @belongsTo
  classroom: Classroom @belongsTo
  inClassroom: Classroom @belongsTo
  student: Person @belongsTo
  responsible: Person @belongsTo
  relationType: RelationType! @rename(attribute: "relation_type_id")
  status: PreRegistrationStatus!
  observation: String
  fields: [PreRegistrationField!]! @hasMany
  stage: ProcessStage @belongsTo
  external: ExternalPerson @hasOne
  linked: [PreRegistrationLinkedByEmail!]! @hasMany
}

type PreRegistrationField {
  id: ID!
  preregistration: PreRegistration! @belongsTo
  field: Field! @belongsTo
  value: String
}

type PreRegistrationLinkedByEmail {
  id: ID!
  preregistration: PreRegistration! @belongsTo
  email: String!
}

extend type Mutation @guard(with: ["web", "sanctum", "prematricula"]) {

  newPreRegistration(
    input: PreRegistrationInput! @spread
  ): [PreRegistration!]!

  sendProtocolsByEmail(
    preregistrations: [ID!]!
    email: String!
  ): Boolean!

  keepOnTheWaitingList(
    id: ID!
    grade: ID
  ): Boolean!

  updateStudentInExternalSystem(
    preregistration: ID!
    cpf: Boolean
    rg: Boolean
    birthCertificate: Boolean
    name: Boolean
    dateOfBirth: Boolean
    gender: Boolean
    phone: Boolean
    mobile: Boolean
    address: Boolean
  ): Boolean!

  savePreRegistrationLinkedByEmail(
    preregistration: ID! @rename(attribute: "preregistration_id")
    email: String!
  ): PreRegistrationLinkedByEmail!

  deletePreRegistrationLinkedByEmail(
    id: ID! @whereKey
  ): PreRegistrationLinkedByEmail! @delete

}

extend type Mutation @guard {
  acceptPreRegistrations(
    ids: [ID!]!
    classroom: ID!
  ): [PreRegistration!]!

  rejectPreRegistrations(
    ids: [ID!]!
    justification: String
  ): [PreRegistration!]!

  summonPreRegistrations(
    ids: [ID!]!
    justification: String
  ): [PreRegistration!]!

  returnToWaitPreRegistrations(
    ids: [ID!]!
  ): [PreRegistration!]!

  updateStudent(
    protocol: String!
    fields: [PreRegistrationFieldInput!]!
  ): PreRegistration

  updateResponsible(
    protocol: String!
    fields: [PreRegistrationFieldInput!]!
  ): PreRegistration

  updatePreRegistration(
    protocol: String!
    grade: ID!
    school: ID!
    period: ID!
  ): PreRegistration

  updateAddress(
    protocol: String!
    address: AddressInput
  ): PreRegistration
}

input PreRegistrationFieldInput {
  field: ID!
  value: String
}

input AddressInput {
  postalCode: String! @rename(attribute: "postal_code")
  address: String!
  number: String!
  complement: String
  neighborhood: String!
  city: String
  stateAbbreviation: String
  cityIbgeCode: Int
  lat: Float! @rename(attribute: "latitude")
  lng: Float! @rename(attribute: "longitude")
  manualChangeLocation: Boolean! @rename(attribute: "manual_change_location")
}

input PreRegistrationWaitingListInput {
  period: ID! @rename(attribute: "period_id")
  school: ID @rename(attribute: "school_id")
}

input PreRegistrationInput {
  type: PreRegistrationType!  @rename(attribute: "preregistration_type_id")
  process: ID! @rename(attribute: "process_id")
  stage: ID! @rename(attribute: "process_stage_id")
  grade: ID! @rename(attribute: "grade_id")
  period: ID! @rename(attribute: "period_id")
  school: ID @rename(attribute: "school_id")
  optionalSchool: ID
  optionalPeriod: ID
  address: AddressInput!
  optionalAddress: AddressInput
  relationType: RelationType! @rename(attribute: "relation_type_id")
  responsible: [PreRegistrationFieldInput!]!
  student: [PreRegistrationFieldInput!]!
  externalPerson: ID @rename(attribute: "external_person_id")
  waitingList: [PreRegistrationWaitingListInput!]
}

type LegacyStudent {
  id: ID!
  name: String!
  cpf: String
  dateOfBirth: String @rename(attribute: "date_of_birth")
  responsibleName: String @rename(attribute: "responsible_name")
  responsibleCpf: String @rename(attribute: "responsible_cpf")
  responsibleDateOfBirth: String @rename(attribute: "responsible_date_of_birth")
  responsiblePhone: String @rename(attribute: "responsible_phone")
}
