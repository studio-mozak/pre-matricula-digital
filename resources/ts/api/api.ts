import axios, { AxiosInstance } from 'axios';

const token = window.config ? window.config.token : '';

export const http: AxiosInstance = axios.create({
  baseURL: String(import.meta.env.VITE_APP_API_URL || ''),
  headers: {
    Authorization: `Token ${token}`,
  },
});

axios.interceptors.response.use(
  (res) => {
    /**
     * If the response from the server has the "errors" object or the object "data" is empty,
     * here it is forced to go to the catch block automatically.
     */
    if (
      (res.data.errors || Object.keys(res.data).length === 0) &&
      !(res.data instanceof Blob)
    ) {
      return Promise.reject({
        response: {
          data: res.data,
        },
      });
    }

    return res;
  },
  (err) => Promise.reject(err)
);

export const rest = <T>(
  type: 'get' | 'post' | 'put' | 'delete',
  url: string,
  params?: object
): Promise<T> => http[type](url, params);

export const graphql = <T>(data: object, config?: object): Promise<T> =>
  axios.post('/graphql', data, {
    headers: {
      Authorization: `Bearer ${token}`,
    },
    ...config,
  });
