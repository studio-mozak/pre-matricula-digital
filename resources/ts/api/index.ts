export * from './api';
export * as City from './services/graphql/city';
export * as Resource from './services/graphql/resource';
