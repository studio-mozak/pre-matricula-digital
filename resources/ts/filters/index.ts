import { Address, GoogleMapsMarker } from '@/types';
import {
  getFormattedDate,
  getFormattedDateTime,
  getFormattedTime,
} from '@/datetime';
import {
  joinAddress,
  joinAddressLine1,
  joinAddressLine2,
  joinAddressLine3,
  markerAddress,
} from '@/util';

export interface Filters {
  formatDateTime: (date: string) => string;
  formatDate: (date: string) => string;
  formatTime: (date: string) => string;
  joinAddress: (address: Address) => string;
  joinAddressLine1: (address: Address) => string;
  joinAddressLine2: (address: Address) => string;
  joinAddressLine3: (address: Address) => string;
  markerAddress: () => GoogleMapsMarker;
}

const filters: Filters = {
  formatDateTime(date: string) {
    return getFormattedDateTime(date);
  },
  formatDate(date: string) {
    return getFormattedDate(date);
  },
  formatTime(date: string) {
    return getFormattedTime(date);
  },
  joinAddress,
  joinAddressLine1,
  joinAddressLine2,
  joinAddressLine3,
  markerAddress,
};

export default filters;
