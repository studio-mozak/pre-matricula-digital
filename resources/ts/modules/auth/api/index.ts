/**
 * Exporting the Axios Instance
 */
export { http, graphql } from '@/api';

/**
 * Exporting all modules
 */
export * as AuthRest from '@/modules/auth/api/services/rest/auth';
