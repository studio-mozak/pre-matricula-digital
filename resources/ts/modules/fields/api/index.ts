/**
 * Exporting the Axios Instance
 */
export { http, graphql } from '@/api';

/**
 * Exporting all modules
 */
export * as Field from './services/graphql/field';
