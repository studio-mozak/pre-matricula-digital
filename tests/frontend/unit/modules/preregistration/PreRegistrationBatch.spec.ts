import '@/plugin/sortby';
import { VueWrapper, flushPromises, mount } from '@vue/test-utils';
import { expect, vi } from 'vitest';
import { useRoute, useRouter } from 'vue-router';
import {Nullable, Option} from '@/types';
import { PreRegistrationBatchProps } from '@/modules/preregistration/types';
import PreRegistrationBatch from '@/modules/preregistration/components/PreRegistrationBatch.vue';
import { Preregistration as PreregistrationApi } from '@/modules/preregistration/api';
import filters from '@/filters';

const spyPush = vi.fn();

vi.mock('@/modules/preregistration/api');
vi.mock('vue-router/dist/vue-router.mjs', () => ({
  useRouter: () => ({
    push: spyPush,
  }),
  useRoute: () => ({
    params: {
      protocol: 'HLYN3T',
    },
  }),
}));

const propsMock: PreRegistrationBatchProps = {
  filter: {
    canAcceptInBatch: false,
    grade: 'string',
    period: 'string',
    process: 'string',
    school: 'string',
    schools: [],
    search: 'string',
    sort: 'string',
    status: 'string',
    type: 'string',
    year: 2021,
    processes: [],
  },
  preregistrations: ['1'],
  step: 'SUMMON',
  processYear: 2022,
  modelValue: true,
};

const getClassroomsByPreregistrationMock: Option[] = [
  {
    key: 1,
    label: 'string',
  },
];

const onMounted = async (
  getClassroomsByPreregistration = getClassroomsByPreregistrationMock,
  props = propsMock
) => {
  vi.mocked(
    PreregistrationApi.getClassroomsByPreregistration
  ).mockResolvedValueOnce(getClassroomsByPreregistration);

  const wrapper = mount(PreRegistrationBatch, {
    props,
    global: {
      provide: {
        $filters: filters,
      },
      stubs: ['x-field', 'modal'],
    },
  });

  await wrapper.vm.$nextTick();
  await flushPromises();

  return wrapper;
};

describe('PreRegistrationBatch', () => {
  let wrapper: VueWrapper<InstanceType<typeof PreRegistrationBatch>>;

  beforeEach(async () => {
    wrapper = await onMounted();

    await wrapper.vm.$nextTick();
  });

  afterEach(() => {
    vi.clearAllMocks();
  });

  test('the component exists', () => {
    expect(wrapper).toBeTruthy();
  });

  test('should match snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  describe('accept', () => {
    test('should return with success response', async () => {
      vi.mocked(PreregistrationApi.postAcceptBatch).mockResolvedValueOnce({
        acceptPreRegistrations: [
          {
            id: '1',
            student: {
              name: 'Aluno 01',
            },
          },
          {
            id: '2',
            student: {
              name: 'Aluno 02',
            },
          },
        ],
      });

      wrapper.vm.classroom = 123;

      wrapper.vm.accept();

      await wrapper.vm.$nextTick();
      await flushPromises();

      expect(wrapper.vm.success).toBeTruthy();
      expect(wrapper.vm.total).toBe(2);
      expect(wrapper.vm.loading).toBeFalsy();
    });

    test('should return with fail response', async () => {
      const errors = [
        {
          message: 'Algo deu errado.',
          extensions: {
            message: 'Erro ao invocar a função.',
          },
        },
      ];

      vi.mocked(PreregistrationApi.postAcceptBatch).mockRejectedValueOnce({
        response: {
          data: {
            errors: errors,
            data: {
              acceptPreRegistrations: [],
            },
          },
        },
      });

      wrapper.vm.classroom = 123;

      wrapper.vm.accept();

      await wrapper.vm.$nextTick();
      await flushPromises();

      expect(wrapper.vm.showError).toBeTruthy();
      expect(wrapper.vm.errorTitle).toBe(
        'Não foi possível realizar o deferimento de todos os(as) alunos(as)'
      );
      expect(wrapper.vm.error).toBe(errors[0].extensions.message);
      expect(wrapper.vm.loading).toBeFalsy();
    });

    test('should show empty classroom message if none classroom was provided', async () => {
      const errors = [
        {
          message: 'Algo deu errado.',
          extensions: {
            message: 'Erro ao invocar a função.',
          },
        },
      ];

      vi.mocked(PreregistrationApi.postAcceptBatch).mockRejectedValueOnce({
        response: {
          data: {
            errors: errors,
            data: {
              acceptPreRegistrations: [],
            },
          },
        },
      });

      wrapper.vm.accept();

      await wrapper.vm.$nextTick();
      await flushPromises();

      expect(wrapper.vm.showError).toBeTruthy();
      expect(wrapper.vm.errorTitle).toBe(
        'É necessário selecionar uma turma para realizar o Deferimento da pré-matrícula.'
      );
      expect(wrapper.vm.error).toBe('');
      expect(wrapper.vm.loading).toBeFalsy();
    });
  });

  describe('summon', () => {
    test('should return with success response', async () => {
      vi.mocked(PreregistrationApi.postSummonBatch).mockResolvedValueOnce({
        summonPreRegistrations: [
          {
            id: '1',
            student: {
              name: 'Aluno 01',
            },
          },
          {
            id: '2',
            student: {
              name: 'Aluno 02',
            },
          },
        ],
      });

      wrapper.vm.summon();

      await wrapper.vm.$nextTick();
      await flushPromises();

      expect(wrapper.vm.success).toBeTruthy();
      expect(wrapper.vm.total).toBe(2);
      expect(wrapper.vm.loading).toBeFalsy();
    });

    test('should return with fail response', async () => {
      const errors = [
        {
          message: 'Algo deu errado.',
          extensions: {
            message: 'Erro ao invocar a função.',
          },
        },
      ];

      vi.mocked(PreregistrationApi.postSummonBatch).mockRejectedValueOnce({
        response: {
          data: {
            errors: errors,
            data: {
              summonPreRegistrations: [],
            },
          },
        },
      });

      wrapper.vm.summon();

      await wrapper.vm.$nextTick();
      await flushPromises();

      expect(wrapper.vm.showError).toBeTruthy();
      expect(wrapper.vm.errorTitle).toBe(
        'Não foi possível realizar a convocação de todos os(as) responsáveis(as)'
      );
      expect(wrapper.vm.error).toBe(errors[0].extensions.message);
      expect(wrapper.vm.loading).toBeFalsy();
    });
  });

  describe('reject', () => {
    test('should return with success response', async () => {
      vi.mocked(PreregistrationApi.postRejectBatch).mockResolvedValueOnce({
        rejectPreRegistrations: [
          {
            id: '1',
            student: {
              name: 'Aluno 01',
            },
          },
          {
            id: '2',
            student: {
              name: 'Aluno 02',
            },
          },
        ],
      });

      wrapper.vm.reject();

      await wrapper.vm.$nextTick();
      await flushPromises();

      expect(wrapper.vm.success).toBe(true);
      expect(wrapper.vm.total).toBe(2);
      expect(wrapper.vm.loading).toBeFalsy();
    });

    test('should return with fail response', async () => {
      const errors = [
        {
          message: 'Algo deu errado.',
          extensions: {
            message: 'Erro ao invocar a função.',
          },
        },
      ];

      vi.mocked(PreregistrationApi.postRejectBatch).mockRejectedValueOnce({
        errors: errors,
      });

      wrapper.vm.reject();

      await wrapper.vm.$nextTick();
      await flushPromises();

      expect(wrapper.vm.showError).toBeTruthy();
      expect(wrapper.vm.errorTitle).toBe('Algo deu errado.');
      expect(wrapper.vm.error).toBe(errors[0].extensions.message);
      expect(wrapper.vm.loading).toBeFalsy();
    });
  });

  describe('getClassroomsByPreregistration', () => {
    test('should call the getClassroomsByPreregistration method with the correct params', async () => {
      vi.mocked(
        PreregistrationApi.getClassroomsByPreregistration
      ).mockResolvedValueOnce(getClassroomsByPreregistrationMock);
      wrapper.vm.getClassroomsByPreregistration();

      await wrapper.vm.$nextTick();
      await flushPromises();

      expect(wrapper.vm.classrooms).toMatchObject(
        getClassroomsByPreregistrationMock
      );
    });
  });
});
