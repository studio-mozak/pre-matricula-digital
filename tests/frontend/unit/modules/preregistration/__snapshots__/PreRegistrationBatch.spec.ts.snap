// Vitest Snapshot v1

exports[`PreRegistrationBatch > should match snapshot 1`] = `
<modal-stub
  bodyclass=""
  fullwidth="false"
  iconleft=""
  iconright=""
  initialloading="false"
  isoverflowvisible="false"
  large="true"
  loadinginputs="false"
  loadingtext="false"
  maximized="false"
  modelvalue="true"
  nofooter="true"
  noheader="false"
  overflowvisible="false"
  persistent="false"
  position="standard"
  seamless="false"
  title="Convocação em lote"
  titleclass=""
  tooltipicon=""
/>
`;
